frontend/src/components/regions/CardDetailsModal/CardDetailsModal.test.js

```diff
@@ -0,0 +1,90 @@
+import { canSetDueTime } from "../../widgets/utils";
+import card from "../../../stories/fixtures/agileCard.json";
+import authUser from "../../../stories/fixtures/authUser.json";
+
+/* Instance 1: Card belongs to current user and there is no due time set */
+const cardAndUserObject1 = {
+    agileCard: {
+        assigneeNames: ["babalwa.mbolekwa@umuzi.org"],
+        dueTime: null,
+    },
+    currentUser: {
+        email: "babalwa.mbolekwa@umuzi.org",
+    }
+}
+
+test("canSetDueTime function returns true if card belongs to current user and there's no due time set", () => {
+    expect(canSetDueTime(cardAndUserObject1.agileCard, cardAndUserObject1.currentUser)).toBe(true);
+});
```

Comment:
The first test would be better structured like so:

```
test("canSetDueTime function returns true if card belongs to current user and there's no due time set", () => {
    // all the data that is used in the test is IN the test

    const user = {id: 1}
    const card = {
        assignees: [user.id],  // there is no space for typos here. It is very explicit. Ids are best when comparing things
        dueTime: null,
    }
    // also note how simple the data is. Flat is better than nested.

    expect(canSetDueTime({card, user})).toBe(true);  
    // Notice the {} in the parameters. Generally it's better to use destructuring instead of having positional arguments. It is much less likely to cause bugs later
});
```